compose/service/dal_interfaces.go

```diff
@@ -12,7 +12,7 @@ type (
 		SearchModels(ctx context.Context) (out dal.ModelSet, err error)
 		ReplaceModel(ctx context.Context, model *dal.Model) (err error)
 		RemoveModel(ctx context.Context, connectionID, ID uint64) (err error)
-		ReplaceModelAttribute(ctx context.Context, model *dal.Model, old, new *dal.Attribute, trans ...dal.TransformationFunction) (err error)
+		ReplaceModelAttribute(ctx context.Context, model *dal.Model, diff *dal.ModelDiff,hasRecords bool, trans ...dal.TransformationFunction) (err error)
```

Comment:
Could this `hasRecords` be a part of `ModelDiff`? 
Could we imply that records exist by inspecting the length of the trans slice?
Could trans be part of ModelDiff?

I'm pretty sure @tjerman will have a comment on this as well

Reply:
Semantically I would say neither "having records" nor "a transformation function" are part of a model diff.
We do need those things to run the "replace model attribute" action. OTOH it does look to me like the ReplaceModelAttribute is getting crowded. I'll wait for @tjerman's feedback before changing anything here.